handlers.go

```diff
@@ -38,6 +38,25 @@ func handleApplications(add func(Change), services map[string]*charm.Application
 			charms[application.Charm] = change.Id()
 		}
 
+		resources := make(map[string]int)
+		localResources := make(map[string]string)
+		for resName, res := range application.Resources {
```

Comment:
Here we keep in mind that validation is done elsewhere, so we trust that either int or string will work.